### ISSUE TITLE
Using with pg-mem?

### ISSUE THREAD

**Issue by dforsber** [oguimbal/pg-server#2]
Would you mind giving an example on how to create pg-server proxy against pg-mem?

I tried with passing `() => newDb().adapters.createPgNative()` for `createSimpleProxy`, but it gives an error when trying to call `setNoDelay()`.

**Comment by oguimbal** [comment]
This is not yet implemented 😑

It requires a refactoring on pg-mem side, which I started, but I did not have the time to finish.

I dont have much time to work on OSS these days, unfortunately. That might change in the future, but I cant tell exactly when this will be supported

**Comment by Asaf-S** [comment]
Hey @oguimbal, thx for all of your hard work. This is a very wanted feature, please find the time to complete it or at least draft an orderly request for help...

**Comment by oguimbal** [comment]
Hi @Asaf-S, preliminary work on [pg-mem](https://github.com/oguimbal/pg-mem) is still ongoing (little by little... I'm dedicating time to OSS when I can, which is not that often).

Any help is always welcome, of course (I'm often receiving PRs on [pg-mem](https://github.com/oguimbal/pg-mem) and [pgsql-ast-parser](https://github.com/oguimbal/pgsql-ast-parser)).

The feature that is needed to make this work is pretty hard to implement, though: It is statement preparation & parameters binding.
Long story short: Work is ongoing (parameterized statements parsing & some preliminary work on the executor have been finalized), but it is still requiring a bit of refactoring.

Rest assured that I still have this issue in mind when working on pg-mem (mainly to support connecting Prisma to pg-mem)

**Comment by Foxious** [comment]
has there been any progress on this? I'd absolutely love to use this for testing prisma

**Comment by rkristelijn** [comment]
+1

how can I help? How far did you come? Which specific branches would be WIP on this topic for `pg-server` and `pg-mem`?

I see https://github.com/oguimbal/pg-mem/issues/360 should be connected.

anyway this is how far I came setting up a stand-alone database server; it doesn't respond ok to a connect command:

https://github.com/rkristelijn/pg-mem-server

**Comment by rkristelijn** [comment]
@oguimbal can we please discuss how to move forward on this issue?

E.g. could you share your work so far in a WIP merge request, so I can see what the solution direction is?
